lib/find_non_repeated_letter.rb

```diff
@@ -9,6 +9,28 @@
 
 
 def find_first_non_repeated_letter(str)
-  #write your method here
-  return true
+  
+  array = str.chars
+  
+  final = array.length - 1
+  
+  answer = ""
+  
+  array.each_index do |i|
+    if i == 0 && array[i] != array[(i+1)]
```

Comment:
I'm not saying a loop isn't ever needed–I'm saying that this particular check for the first character doesn't need to be part of that loop.

You've got

```ruby
if i == 0 && array[i] != array[(i+1)]
```

so if that line of code runs, `i` is **definitely** `0`. Which means you could re-write it as

```ruby
if i == 0 && array[0] != array[(0+1)]
```

So while some of the operations in your loop *do* depend on the changing value of `i`, this one doesn't. It's only ever run when `i` is `0`.